go-controller/pkg/node/openflow_manager.go

```diff
@@ -245,7 +245,38 @@ func (c *openflowManager) updateBridgeFlowCache(hostIPs []net.IP, hostSubnets []
 	}
 	dftFlows = append(dftFlows, dftCommonFlows...)
 
-	c.updateFlowCacheEntry("NORMAL", []string{fmt.Sprintf("table=0,priority=0,actions=%s\n", util.NormalAction)})
+	if config.OvnKubeNode.Mode == types.NodeModeDPU {
+		// Retrieve the local_ip for the tunnel endpoint which will be used to create OpenFlow actions
+		checkCmd := []string{
+			"get",
+			"Open_vSwitch",
+			".",
+			"external_ids:ovn-encap-ip",
+		}
+		encapIP, _, err := util.RunOVSVsctl(checkCmd...)
+		if err != nil {
+			return err
+		}
+		encapIP = strings.TrimSpace(encapIP)
+
+		// The extra actions, compared to the full mode, are due to a DHCP server on the bridge.
+		// While not impacting the other flows negatively it implies something "custom"
+		// this could be a future feature to allow users to add custom flow actions on the managed ovnkube bridge via Kubernetes API.
+		c.updateFlowCacheEntry("NORMAL", []string{
+			// Add normal action on table 0 (act as a switch)
+			fmt.Sprintf("table=0,priority=0,actions=%s\n", util.NormalAction),
+			// Add OpenFlow action with high priority that allows a normal action for DHCP servers
+			// that are serving on the ovnkube managed bridge.
+			fmt.Sprintf("table=0,priority=999,udp,tp_dst=67,actions=%s\n", util.NormalAction),
```

Comment:
So I think the reason you need this flow is because in br-ex when we see IP traffic, we send to table 1 for IP handling. In that table we do not handle broadcast traffic. My understanding of why you are adding the flow is 1 or more clients on the br-ex bridge will send DHCP discover/request to 255.255.255.255:67. I think it would make more sense to add a generic flow that matches on 255.255.255.255 and does a normal action. This should be enabled in table 1, and should not depend on DPU.

I could see a case where a localnet port VM tries to DHCP Request, and the DHCP server is running on the host. This would also solve that case I think. wdyt @ricky-rav ?

Reply:
yes I think there are a few things that will fix your traffic pattern:
1. https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5313
2. https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5334
3. One more PR to match broadcast traffic and flood it.
4. The earlier PRs may not take into account that in DPU mode, ofportHost should be used and not hardcode LOCAL. Might need to fix this.

I'll be on PTO next week, but when I get back I can probably just deploy kind and add a dummy interface to ovs as my "PF" and then run ovnk in DPU mode. Then I can see the flows and I will know what we need to fix. If you could also send me a ovs-ofctl dump-flows br-ovn on your side, or give me tmate access to a cluster when I get back that would also be helpful.